src/vfio_device.rs

```diff
@@ -121,27 +122,31 @@ struct vfio_region_info_with_cap {
 /// address translation mapping tables.
 pub struct VfioContainer {
     container: File,
+    device_fd: Arc<DeviceFd>,
+    groups: Mutex<HashMap<u32, Arc<VfioGroup>>>,
 }
 
 impl VfioContainer {
     /// Create a container wrapper object.
-    pub fn new() -> Result<Self> {
+    ///
+    /// # Arguments
+    /// * `device_fd`: file handle of the KVM VFIO device.
+    pub fn new(device_fd: Arc<DeviceFd>) -> Result<Self> {
         let container = OpenOptions::new()
             .read(true)
             .write(true)
             .open("/dev/vfio/vfio")
             .map_err(VfioError::OpenContainer)?;
 
-        Ok(VfioContainer { container })
-    }
-
-    /// Initialize the IOMMU backend with groups attached to the container.
-    pub fn initialize_iommu(&self) -> Result<()> {
-        self.check_api_version()?;
-        self.check_extension(VFIO_TYPE1v2_IOMMU)?;
-        self.set_iommu(VFIO_TYPE1v2_IOMMU)?;
+        let obj = VfioContainer {
```

Comment:
Nit: Could we call that one `container` instead of `obj`?

Reply:
Fixed:)